### ISSUE TITLE
Uninstall option

### ISSUE THREAD

**Issue by jcolesio** [mintty/wsltty#15]
Any plans to include an uninstaller in order to revert back to default installation?

**Comment by mintty** [comment]
Not sure what a "default installation" is. The next release of wsltty (in a few days) will install a Start Menu sub-folder, with setup options to install and uninstall the Explorer context menu items. I'm still thinking whether to install a desktop icon at all, but it can easily be removed anyway. See #13.
A complete uninstall option is not planned because WSL itself doesn't have one either.

**Comment by mintty** [comment]
Released install/uninstall for Explorer context menu items with 0.7.0.

**Comment by MicahZoltu** [comment]
If you won't be releasing an uninstaller, can you please list _all_ of the things that need to be done to remove wsltty?  It is causing severe problems on my machine and I need to delete it, but I don't know what registry entries it has written, folders it has created, etc.

**Comment by mintty** [comment]
The new Store-based distributions have an uninstall option, right? (The previous Bash on Windows didn't, so wsltty didn't have one either.) So I'll improve on this.

For the records: Software is installed in %LOCALAPPDATA%/wsltty, configuration directory is set up in %APPDATA%/wsltty.
Registry entries are only used for the Windows Explorer context menu integration, for which an uninstaller exists already, albeit with a problem (https://github.com/mintty/wsltty/issues/52#issuecomment-318851898).

**Comment by mintty** [comment]
Here is the deal: if #63 gets resolved thanks to some kind person's contribution, I will take care of a proper uninstallation feature.

**Comment by mintty** [comment]
A Windows Appx package is now available in the [wsltty.appx](https://github.com/mintty/wsltty.appx/) repository.
It includes an Uninstall option. Closing this issue.

**Comment by R4meau** [comment]
@mintty Well, cheers for the Appx package but I still can't fully uninstall the old WSLtty. I deleted the folders located at %LOCALAPPDATA%/wsltty and %APPDATA%/wsltty but the Explorer context menu entry still persists. How do you suggest I fix this?

**Comment by mintty** [comment]
Reinstall (either version, may be the appx package) and from the Start menu WSLtty subfolder, run "remove from context menu".

**Comment by Biswa96** [comment]
Universal Widows Packages aka. Appx packages has no code to uninstall. Windows has the built-in feature to delete all the associated files during uninstall. Think that feature as 'garbage collection' in managed languages (e.g. Java, C# etc.).

**Comment by R4meau** [comment]
@mintty I reinstalled wsltty, I installed the Appx package, I ran the `remove from context menu` command (as Admin), then I deleted the folders located at `%LOCALAPPDATA%/wsltty` and `%APPDATA%/wsltty`, then I ran `install_appx.cmd` command (as Admin) once again so I can enter `u` to uninstall the Appx package. Finally, I restarted the computer.

They are both still in the context menu and the Appx version is still installed. Am I missing something here?

**Comment by Biswa96** [comment]
Appx package and normal installation are very much different. Also their installation paths are different.

**Comment by R4meau** [comment]
@Biswa96 I was just following @mintty's suggestion.

**Comment by mintty** [comment]
It seems that the current version of the script invocation `config-distros.sh -contextmenu-remove` does not match the context menu entries of early versions of wsltty anymore, due to changed parameters (particularly `--WSL` rather than `--wsl`). Please use the registry editor (`regedit`) to navigate to HKEY_CURRENT_USER\Software\Classes\Directory\shell and HKEY_CURRENT_USER\Software\Classes\Directory\Background and find all entries referring to mintty or WSL. What's a value of them?
These entries can be removed manually.

**Comment by mintty** [comment]
Please navigate to HKEY_CURRENT_USER\Software\Classes\Directory, right-click on it and export the registry subtree. Upload the file here.

**Comment by R4meau** [comment]
Here is the content of that registry:
```
Windows Registry Editor Version 5.00

[HKEY_CURRENT_USER\Software\Classes\Directory]

[HKEY_CURRENT_USER\Software\Classes\Directory\Background]

[HKEY_CURRENT_USER\Software\Classes\Directory\Background\shell]

[HKEY_CURRENT_USER\Software\Classes\Directory\Background\shell\Hyper]
@="Open Hyper here"
"Icon"="\"C:\\Users\\Nick\\AppData\\Local\\hyper\\app-2.0.0\\Hyper.exe\""

[HKEY_CURRENT_USER\Software\Classes\Directory\Background\shell\Hyper\command]
@="\"C:\\Users\\Nick\\AppData\\Local\\hyper\\app-2.0.0\\Hyper.exe\" \"%V\""

[HKEY_CURRENT_USER\Software\Classes\Directory\Background\shell\VSCode]
@=hex(2):4f,00,70,00,65,00,6e,00,20,00,77,00,26,00,69,00,74,00,68,00,20,00,43,\
  00,6f,00,64,00,65,00,00,00
"Icon"=hex(2):43,00,3a,00,5c,00,55,00,73,00,65,00,72,00,73,00,5c,00,4e,00,69,\
  00,63,00,6b,00,5c,00,41,00,70,00,70,00,44,00,61,00,74,00,61,00,5c,00,4c,00,\
  6f,00,63,00,61,00,6c,00,5c,00,50,00,72,00,6f,00,67,00,72,00,61,00,6d,00,73,\
  00,5c,00,4d,00,69,00,63,00,72,00,6f,00,73,00,6f,00,66,00,74,00,20,00,56,00,\
  53,00,20,00,43,00,6f,00,64,00,65,00,5c,00,43,00,6f,00,64,00,65,00,2e,00,65,\
  00,78,00,65,00,00,00

[HKEY_CURRENT_USER\Software\Classes\Directory\Background\shell\VSCode\command]
@=hex(2):22,00,43,00,3a,00,5c,00,55,00,73,00,65,00,72,00,73,00,5c,00,4e,00,69,\
  00,63,00,6b,00,5c,00,41,00,70,00,70,00,44,00,61,00,74,00,61,00,5c,00,4c,00,\
  6f,00,63,00,61,00,6c,00,5c,00,50,00,72,00,6f,00,67,00,72,00,61,00,6d,00,73,\
  00,5c,00,4d,00,69,00,63,00,72,00,6f,00,73,00,6f,00,66,00,74,00,20,00,56,00,\
  53,00,20,00,43,00,6f,00,64,00,65,00,5c,00,43,00,6f,00,64,00,65,00,2e,00,65,\
  00,78,00,65,00,22,00,20,00,22,00,25,00,56,00,22,00,00,00

[HKEY_CURRENT_USER\Software\Classes\Directory\Background\shellex]

[HKEY_CURRENT_USER\Software\Classes\Directory\Background\shellex\ContextMenuHandlers]

[HKEY_CURRENT_USER\Software\Classes\Directory\Background\shellex\ContextMenuHandlers\ FileSyncEx]
@="{CB3D0F55-BC2C-4C1A-85ED-23ED75B5106B}"

[HKEY_CURRENT_USER\Software\Classes\Directory\shell]

[HKEY_CURRENT_USER\Software\Classes\Directory\shell\VSCode]
@=hex(2):4f,00,70,00,65,00,6e,00,20,00,77,00,26,00,69,00,74,00,68,00,20,00,43,\
  00,6f,00,64,00,65,00,00,00
"Icon"=hex(2):43,00,3a,00,5c,00,55,00,73,00,65,00,72,00,73,00,5c,00,4e,00,69,\
  00,63,00,6b,00,5c,00,41,00,70,00,70,00,44,00,61,00,74,00,61,00,5c,00,4c,00,\
  6f,00,63,00,61,00,6c,00,5c,00,50,00,72,00,6f,00,67,00,72,00,61,00,6d,00,73,\
  00,5c,00,4d,00,69,00,63,00,72,00,6f,00,73,00,6f,00,66,00,74,00,20,00,56,00,\
  53,00,20,00,43,00,6f,00,64,00,65,00,5c,00,43,00,6f,00,64,00,65,00,2e,00,65,\
  00,78,00,65,00,00,00

[HKEY_CURRENT_USER\Software\Classes\Directory\shell\VSCode\command]
@=hex(2):22,00,43,00,3a,00,5c,00,55,00,73,00,65,00,72,00,73,00,5c,00,4e,00,69,\
  00,63,00,6b,00,5c,00,41,00,70,00,70,00,44,00,61,00,74,00,61,00,5c,00,4c,00,\
  6f,00,63,00,61,00,6c,00,5c,00,50,00,72,00,6f,00,67,00,72,00,61,00,6d,00,73,\
  00,5c,00,4d,00,69,00,63,00,72,00,6f,00,73,00,6f,00,66,00,74,00,20,00,56,00,\
  53,00,20,00,43,00,6f,00,64,00,65,00,5c,00,43,00,6f,00,64,00,65,00,2e,00,65,\
  00,78,00,65,00,22,00,20,00,22,00,25,00,56,00,22,00,00,00

[HKEY_CURRENT_USER\Software\Classes\Directory\shellex]

[HKEY_CURRENT_USER\Software\Classes\Directory\shellex\ContextMenuHandlers]

[HKEY_CURRENT_USER\Software\Classes\Directory\shellex\ContextMenuHandlers\ FileSyncEx]
@="{CB3D0F55-BC2C-4C1A-85ED-23ED75B5106B}"
```

**Comment by mintty** [comment]
Actually, there is also registry subtree `HKEY_CLASSES_ROOT\Directory` with subfolders `Background` and `shell`. While the wsltty installer does not touch that hierarchy, the wsltty.appx package might do.
@Biswa96, can you comment on this?
If that's the cause of your context menu entries, just delete the respective ones there, please.

**Comment by mintty** [comment]
Any occurrence of "mintty" in the registry? This is weird. Sure you really still have the entries in the Explorer context menu right now?

**Comment by R4meau** [comment]
I did a full search for the term `tty` and deleted everything I found - except for the ones that are not related to terminals - until I reached the end of the registry - I found a lot btw. Then I restarted. It's still in my Explorer context menu.

**Comment by mintty** [comment]
Does anyone have the same issue?
Can anyone comment on this?

**Comment by mintty** [comment]
Released 3.0.5.
